File: Fase 2/Evidencias Proyecto/Evidencias de sistema/Aplicación/api/app/schemas/cesfam.py
from pydantic import BaseModel

# -------- Create / Update / Out --------
class CesfamCreate(BaseModel):
    id_comuna: int
    nombre_cesfam: str
    telefono: int
    direccion: str
    email: str
    estado: bool

class CesfamUpdate(BaseModel):
    id_comuna: int | None = None
    nombre_cesfam: str | None = None
    telefono: int | None = None
    direccion: str | None = None
    email: str | None = None
    estado: bool | None = None

class CesfamOut(BaseModel):
    id_cesfam: int
    id_comuna: int
    nombre_cesfam: str
    telefono: int
    direccion: str
    email: str
    estado: bool
    class Config:
        from_attributes = True

# -------- List filters (paginación) --------
class CesfamListFilters(BaseModel):
    page: int = 1
    page_size: int = 20
    id_comuna: int | None = None
    estado: bool | None = True  # por defecto, solo activos

# -------- Habilitar / Deshabilitar --------
class CesfamSetEstado(BaseModel):
    habilitar: bool

File: Fase 2/Evidencias Proyecto/Evidencias de sistema/Aplicación/CuidaSalud/src/services/cesfam.ts
// src/services/cesfam.ts
const API_HOST = "http://127.0.0.1:8000";
const RUTA_CESFAM = `${API_HOST}/cesfam`;

export type CesfamOut = {
  id_cesfam: number;
  id_comuna: number;
  nombre_cesfam: string;
  telefono: number | null;
  direccion: string | null;
  email: string | null;
  estado: boolean;
};

export type Page<T> = {
  items: T[];
  total: number;
  page: number;
  page_size: number;
};

type ApiResult<T> =
  | { ok: true; data: T }
  | { ok: false; status: number; message: string; details?: any };

export async function listCesfam(params?: {
  page?: number;
  page_size?: number;
  id_comuna?: number;
  estado?: boolean | null; // por defecto true en la API
}) {
  const page = params?.page ?? 1;
  const page_size = params?.page_size ?? 5000;
  const id_comuna = params?.id_comuna;
  const estado = params?.estado ?? true;

  const url = new URL(RUTA_CESFAM);
  url.searchParams.set("page", String(page));
  url.searchParams.set("page_size", String(page_size));
  if (id_comuna != null) url.searchParams.set("id_comuna", String(id_comuna));
  if (estado !== null) url.searchParams.set("estado", String(estado));

  try {
    const resp = await fetch(url.toString());
    if (!resp.ok) {
      let message = `Error HTTP ${resp.status}`;
      try {
        const err = await resp.json();
        message = typeof err === "string" ? err : err?.message ?? message;
      } catch {}
      return { ok: false, status: resp.status, message } as ApiResult<never>;
    }
    const data = (await resp.json()) as Page<CesfamOut>;
    return { ok: true, data } as ApiResult<Page<CesfamOut>>;
  } catch (e: any) {
    return { ok: false, status: 0, message: e?.message ?? "Fallo de red" } as ApiResult<never>;
  }
}
